### PR TITLE
feat: add whitepaper, join mainnet, ethereum address on home

### PR DESCRIPTION
**Pull request by ice-orion**
* Add whitepaper, join mainnet, ethereum address on home 
* Update typescript to v5, node to v18, some dependant development libs - that is also going to be a part of RN 0.73 update but currently that was a requirement for `viem` lib
* Update moengage lib, moengage patch so it works with the new TS version

### CODE REVIEW

File: src/constants/links.ts [hunk 1]
```diff
@@ -46,4 +46,6 @@ export const LINKS = {
     'https://www.figma.com/community/file/1256148399031329897/ice-Community-Assets',
   ICE_VIDEOS:
     'https://drive.google.com/drive/u/1/folders/1E_3zVZzs2vqUNgUd05qfi6cJ_axjlvQU',
+  WHITEPAPER: 'http://ice.io/whitepaper',
```
Comment (ice-ares) [inline src/constants/links.ts]: https

File: src/services/ethereum/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import {mainnet} from 'viem/chains';
+
+export const isValidEthereumAddress = (address: string): boolean => {
+  return isAddress(address);
+};
+
+export const isEoaEthereumAddress = async (
+  address: string,
+): Promise<boolean> => {
+  const client = createPublicClient({
```
Comment (ice-ares) [inline src/services/ethereum/index.tsx]: No. Create a global singleton

**Review by ice-atlas — DISMISSED** [review]

File: src/constants/links.ts (hunk 1 above)
Comment (ice-orion) [reply src/constants/links.ts]: fixed

File: src/services/ethereum/index.tsx (hunk 2 above)
Comment (ice-orion) [reply src/services/ethereum/index.tsx]: done

**Review by ice-ares — DISMISSED** [review]

**Review by ice-ares — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Pull request merged**